docs/gallery/plot_external_resources.py

```diff
@@ -73,6 +73,16 @@
 #    This may or may not include a string representing the resource and a colon.
 #    Use the format provided by the resource. For example, Identifiers.org uses the ID ``ncbigene:22353``
 #    but the NCBI Gene uses the ID ``22353`` for the same term.
+# 8. In a majority of cases, :py:class:`~hdmf.common.resources.Object` objects will have an empty string
+#    for 'field'. The :py:class:`~hdmf.common.resources.ExternalResources` class supports compound data_types.
+#    In that case, 'field' would be the field of the compound data_type that has an external reference.
+# 9. The :py:class:`~hdmf.common.resources.Object` object we add to the
+#    :py:class:`~hdmf.common.resources.ObjectTable` has a data_type.
```

Comment:
```suggestion
# 9. The :py:class:`~hdmf.common.resources.Object.object_id` object we add to the
#    :py:class:`~hdmf.common.resources.ObjectTable` has a data_type.
```

Not sure if this is correct edit. Can you clarify what you want to express in rule 9. ``hdmf.common.resources.Object`` has ``object_id``, ``relative_path``, and ``field``. The ``object_id`` refers to an object with a ``data_type`` but the ``relative_path`` will point to an object without a ``data_type``.